src/core/mac/mac.cpp

```diff
@@ -2418,10 +2458,124 @@ Error Mac::SetWakeupChannel(uint8_t aChannel)
     VerifyOrExit(mSupportedChannelMask.ContainsChannel(aChannel), error = kErrorInvalidArgs);
     mWakeupChannel = aChannel;
 
+#if OPENTHREAD_CONFIG_WAKEUP_END_DEVICE_ENABLE
+    UpdateWakeupListening();
+#endif
+
+exit:
+    return error;
+}
+#endif
+
+#if OPENTHREAD_CONFIG_WAKEUP_END_DEVICE_ENABLE
+void Mac::SetWedListenInterval(uint32_t aInterval)
+{
+    mWedListenInterval = aInterval;
+    UpdateWakeupListening();
+}
+
+Error Mac::SetWedListenDuration(uint16_t aDuration)
+{
+    Error error = kErrorNone;
+
+    VerifyOrExit(aDuration >= kMinWedListenDuration, error = kErrorInvalidArgs);
+
+    mWedListenDuration = aDuration;
+    UpdateWakeupListening();
+
+exit:
+    return error;
+}
+
+Error Mac::SetWedListenEnabled(bool aEnable)
+{
+    Error error = kErrorNone;
+
+    VerifyOrExit(GetWedListenInterval() > GetWedListenDuration(), error = kErrorInvalidArgs);
+
+#if OPENTHREAD_CONFIG_MAC_CSL_RECEIVER_ENABLE
+    if (aEnable && IsCslEnabled())
```

Comment:
```suggestion
    if (aEnable && GetCslPeriod() > 0)
```
Otherwise, one can still enable both by running "csl period X" and "wakeup listen enable" before attaching to a parent. Then after attaching to the parent, the wake-up listening will not get disabled.